tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -3882,14 +3679,14 @@ Status FragmentMetadata::load_footer(
   loaded_metadata_.tile_sum_.resize(num, false);
   loaded_metadata_.tile_null_count_.resize(num, false);
 
-  RETURN_NOT_OK(load_generic_tile_offsets(cbuff.get()));
+  load_generic_tile_offsets(*deserializer.get());
 
   loaded_metadata_.footer_ = true;
 
   // If the footer_size is not set lets calculate from how much of the buffer we
   // read
   if (footer_size_ == 0)
-    footer_size_ = cbuff->offset() - offset;
+    footer_size_ = deserializer->offset() - offset;
```

Comment:
I think we don't have to add offset() here... We know the original size, and we can call size() to know what is left. Subtracting the two should give us the size?

Reply:
changes made, offset computed from capture of starting deserializer size and 'current' size at this point